File: Math/lua/Divisor.lua
if #arg < 1 then
    return
end

local n = tonumber(arg[1])
if not n then
    return
end

local max_precision = n * 2
local max_display = 45

local array_x = {}
local array_y = {}
for i = 1, n do
    array_x[i] = 1
    array_y[i] = i
end

local digits = {}

for i = 1, n do
    local x = array_x[i]
    local y = array_y[i]

    local col1 = string.format("[%d]", i)
    local col2 = string.format("[%d/%d]", x, y)

    -- Right alignment: widths 7 and 10
    -- Lua padding: string.rep(" ", width - #s) .. s
    local p1 = string.rep(" ", 7 - #col1) .. col1
    local p2 = string.rep(" ", 10 - #col2) .. col2
    io.write(p1 .. "\t" .. p2)

    local z = math.floor(x / y)
    x = (x % y) * 10

    local head = 0
    local tail = 0
    local repeat_count = 0
    local length = 0

    while tail == 0 or (tail < max_precision and x > 0 and (repeat_count < 10 or repeat_count < length)) do
        digits[tail + 1] = math.floor(x / y)
        x = (x % y) * 10

        if tail > 0 then
            local search = 0
            while search < tail and digits[head + 1] ~= digits[tail + 1] do
                search = search + 1
                head = head + 1
                head = head % tail
                repeat_count = 0
            end

            if head < tail and digits[head + 1] == digits[tail + 1] then
                length = tail - head
                repeat_count = repeat_count + 1
                if repeat_count == 1 then
                    while repeat_count <= head and digits[head - repeat_count + 1] == digits[tail - repeat_count + 1] do
                        repeat_count = repeat_count + 1
                    end
                end
                head = head + 1
            else
                repeat_count = 0
                length = 0
                head = 0
            end
        end
        tail = tail + 1
    end

    io.write("\t" .. tostring(z) .. ".")

    if repeat_count > 0 then
        for j = 1, head - repeat_count do
            if j > max_display then break end
            io.write(tostring(digits[j]))
        end
        io.write("|")
        for j = head - repeat_count + 1, tail - repeat_count do
            if j > max_display then break end
            io.write(tostring(digits[j]))
        end
        if tail - repeat_count < max_display then
            io.write("|")
        else
            io.write("..")
        end
    else
        for j = 1, tail do
            if j > max_display then break end
            io.write(tostring(digits[j]))
        end
    end
    io.write("\n")
end

File: Math/c/Divisor.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char *argv[]) {
    if (argc < 2) {
        return 0;
    }

    int n = atoi(argv[1]);
    if (n <= 0) {
        return 0;
    }

    int max_precision = n * 2;
    int max_display = 45;

    int *array_x = malloc(n * sizeof(int));
    int *array_y = malloc(n * sizeof(int));
    for (int i = 0; i < n; i++) {
        array_x[i] = 1;
        array_y[i] = i + 1;
    }

    int *digits = malloc(max_precision * sizeof(int));

    for (int i = 1; i <= n; i++) {
        int x = array_x[i - 1];
        int y = array_y[i - 1];

        char col1[20], col2[40];
        sprintf(col1, "[%d]", i);
        sprintf(col2, "[%d/%d]", x, y);

        // Right alignment: widths 7 and 10
        printf("%7s\t%10s", col1, col2);

        int z = x / y;
        x = (x % y) * 10;

        int head = 0;
        int tail = 0;
        int repeat = 0;
        int length = 0;

        while (tail == 0 || (tail < max_precision && x > 0 && (repeat < 10 || repeat < length))) {
            digits[tail] = x / y;
            x = (x % y) * 10;

            if (tail > 0) {
                int search = 0;
                while (search < tail && digits[head] != digits[tail]) {
                    search++;
                    head++;
                    head %= tail;
                    repeat = 0;
                }

                if (head < tail && digits[head] == digits[tail]) {
                    length = tail - head;
                    repeat++;
                    if (repeat == 1) {
                        while (repeat <= head && digits[head - repeat] == digits[tail - repeat]) {
                            repeat++;
                        }
                    }
                    head++;
                } else {
                    repeat = 0;
                    length = 0;
                    head = 0;
                }
            }
            tail++;
        }

        printf("\t%d.", z);

        if (repeat > 0) {
            for (int j = 0; j < head - repeat && j < max_display; j++) {
                printf("%d", digits[j]);
            }
            printf("|");
            for (int j = head - repeat; j < tail - repeat && j < max_display; j++) {
                printf("%d", digits[j]);
            }
            if (tail - repeat < max_display) {
                printf("|");
            } else {
                printf("..");
            }
        } else {
            for (int j = 0; j < tail && j < max_display; j++) {
                printf("%d", digits[j]);
            }
        }
        printf("\n");
    }

    free(array_x);
    free(array_y);
    free(digits);

    return 0;
}
